src/services/bootstrap/PreloadData.ts

```diff
@@ -52,21 +52,25 @@ export class PreloadData {
     this.defineEnvironment();
 
     await this.updateUser();
-    await this.updateUserProfile();
-    await this.updateBranding();
-    new ResourceFetcherService().prefetchResources(this.store.getState());
+    await this.updateJsonRpcMasterApi();
 
-    this.updateRestApiClient();
-    this.updateJsonRpcMasterApi();
+    const settings = await this.updateWorkspaceSettings();
 
-    this.updateWorkspaces();
-    this.updateInfrastructureNamespaces();
+    await Promise.all([
+      Promise.all([
+        this.updateBranding(),
+        this.updateWorkspaces(),
+        this.updateInfrastructureNamespaces(),
+        this.updateUserProfile(),
+      ]),
+      Promise.all([
+        this.updatePlugins(settings),
+        this.updateRegistriesMetadata(settings),
+        this.updateDevfileSchema(),
+      ])
+    ]);
```

Comment:
Have you considered using [Promise.allSettled()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled) here? And also I wonder why do you group methods this way?

Reply:
I was thinking about  ```Promise.allSettled()``` here, but I need a promise that rejects after branding data request fails. So, I decide to use ```Promise.all() ``` instead .